app/js/components/orderEntry/addForm/AddForm.jsx

```diff
@@ -1,46 +1,110 @@
-            <li className="nav-item">
-              <a className="nav-link"
-                id="standard-tab"
-                data-toggle="pill"
-                href="#standard"
-                role="tab"
-                aria-controls="standard"
-                aria-selected="false">Standard Dosing
-              </a>
-            </li>
-            <li className="nav-item">
-              <a className="nav-link"
-                id="free-tab"
-                data-toggle="pill"
-                href="#free"
-                role="tab"
-                aria-controls="free"
-                aria-selected="false">Free Text
-              </a>
-            </li>
-          </ul>
-          <br />
-
-          <div className="tab-content" id="pills-tabContent">
-            <FreeText />
-            <StandardDose />
-          </div>
+export class AddForm extends React.Component {
+  state = {
+    drugFormPopulated: false,
+    action: '',
+    fields: {
+      completeInstructions: '',
+      dose: '',
+      dosingUnit: '',
+      frequency: '',
+      route: '',
+      duration: '',
+      durationUnit: '',
+      dispensingUnit: '',
+      dispensingQuantity: '',
+      reason: '',
+      note: '',
+    },
+  };
+  componentDidMount() {
+    this.props.getOrderEntryConfigurations();
+  }
+  handleCancel = () => {
+    // Cancel a draft drug order
+  }
+  handleSubmitDrugForm = () => (
+    this.setState({ drugFormPopulated: true, action: "NEW" })
+  )
+  handleChange = (e) => {
+    this.setState({
+      ...this.state,
+      fields: { ...this.state.fields, [e.target.name]: e.target.value }
+    });
+  }
+  addDrugOrder = (event) => {
+    event.preventDefault();
+    // make post request to API
+  }
+  renderDraftDataTable = () => (
+    <div>
+      <DraftDataTable
+        drugName={this.props.drugName}
+        fields={this.state.fields}
+        status={this.state.action}
+        handleCancel={this.handleCancel}
+        handleSubmit={this.addDrugOrder}
+      />
+    </div>
+  );
+  renderDrugOrderForms = () => (
+    <div>
+      <div>
+        <ul>
+          <li><a href="#standard">Standard Dosing</a></li>
+          <li><a href="#free">Free Text</a></li>
+        </ul>
+        <br />
 
+        <div>
+          <FreeText
+            fields={this.state.fields}
+            dispensingUnits={this.props.dispensingUnits}
+            handleChange={this.handleChange}
+            handleSubmit={this.handleSubmitDrugForm}
+          />
+          <StandardDose />
         </div>
+
+      </div>
+    </div>
+  );
+  render() {
+    return (
+      <div>
+        {
+          this.state.drugFormPopulated ?
+            this.renderDraftDataTable() :
+            this.renderDrugOrderForms()
+        }
       </div>
     );
   }
 }
 
-export default AddForm;
+const mapStateToProps = ({ orderEntryConfigurations }) => ({
+  dispensingUnits: orderEntryConfigurations.configurations,
```

Comment:
The name dispensingUnits seems to be ambigious to the state you are mapping it to. orderEntryConfigurations.configurations also contains other configurations apart from dispensingUnits.